docs: Add more documentation for video transcoding settings. (#13388)
immich-app/immich: docs/docs/administration/system-settings.md

Author: Ok, I've moved the docs to that page. I also expanded a bit to cover most of the Immich-specific settings in there. For now I have left out things that are basically ffmpeg flags.

I used the config file attribute as the heading name, but I'm not sure if this is the best option as it won't make as much sense for people configuring things in the UI. I could use the english names in the headers and mention the config properties in the body instead if that is preferable. I don't know if that would be confusing to people using Immich in other localizations but I AFAICT the docs aren't localized so I don't know if there is a better option.

Reviewer: You can use the English title for them and put the JSON path in parentheses, clarifying at the start that this is what the parentheses mean.

Author: Done.

The default value is `aac`.

### Video Codec (`ffmpeg.targetVideoCodec`) {#ffmpeg.targetVideoCodec}

Which video codec to use when the video stream is being transcoded. Can be one of `h264`, `hevc`, `vp9` or `av1`.

The default value is `h264`.

### Target resolution (`ffmpeg.targetResolution`) {#ffmpeg.targetResolution}

When transcoding video downscale the largest dimension to this value. Videos are never upscaled.

The default value is `720`.

### Transcode policy (`ffmpeg.transcode`) {#ffmpeg.transcode}

The transcoding policy configures which streams of a video asset will be transcoded. The transcoding decision is made independently for video streams and audio streams.

The default policy is `required`.

#### All videos (`all`) {#ffmpeg.transcode-all}

Videos are always transcoded. This ensures consistency during video playback.

#### Don't transcode any videos (`disabled`) {#ffmpeg.transcode-disabled}

Videos are never transcoded. This saves space and resources on the server, but may prevent playback on devices that don't support the source format (especially web browsers) or result in high bandwidth usage when playing high-bitrate files.

> #### Only videos not in an accepted format (`required`) {#ffmpeg.transcode-required}

Author: I've added that property to the docs in a separate section.

Video is transcoded when any of the following conditions are met:

> - The video is HDR.

Author: How should I handle this? That change isn't live yet. I can prepare a PR to merge once that goes in or we can add a note to that PR if it doesn't land before this one.

> - The video is HDR.

Reviewer: It's live now :P

> - The video is HDR.

Author: Added the condition.

- The video codec is not in `acceptedVideoCodecs`.

Audio is transcoded if the audio codec is not in `acceptedAudioCodecs`.

#### `bitrate`

In addition to the conditions in `required`, video is also transcoded if their bitrate is over `maxBitrate`.

#### `optimal`

> In addition to the conditions in `required`, video is also transcoded if the horizontal **and** vertical resolutions are higher than `targetResolution`.

Reviewer: ```suggestion
In addition to the conditions in `required`, video is also transcoded if the horizontal **and** vertical dimensions are higher than `targetResolution`.
```

## Video Transcoding Settings

The system administrator can define parameters according to which video files will be converted to different formats (depending on the settings). The settings can be changed in depth, to learn more about the terminology used here, refer to FFmpeg documentation for [H.264](https://trac.ffmpeg.org/wiki/Encode/H.264) codec, [HEVC](https://trac.ffmpeg.org/wiki/Encode/H.265) codec and [VP9](https://trac.ffmpeg.org/wiki/Encode/VP9) codec.

Which streams of a video file will be transcoded is determined by the [Transcode Policy](#ffmpeg.transcode). Streams that are transcoded use the following settings (config file name in brackets). Streams that are not transcoded are untouched and preserve their original settings.

### Accepted containers (`acceptedContainers`) {#ffmpeg.acceptedContainers}

> If the video asset's container format is not in this list it will be remuxed to MP4 even if no streams need to be transcoded.

Reviewer: ```suggestion
If the video asset's container format is not in this list, it will be remuxed to MP4 even if no streams need to be transcoded.
```

The default set of accepted container formats is `mov`, `ogg` and `webm`.

### Preset (`ffmpeg.preset`) {#ffmpeg.preset}

The amount of "compute effort" to put into transcoding. These use [the preset names from h264](https://trac.ffmpeg.org/wiki/Encode/H.264#Preset) and will be converted to appropriate values for encoders that configure effort in different ways.

The default value is `ultrafast`.

### Audio codec (`ffmpeg.targetAudioCodec`) {#ffmpeg.targetAudioCodec}

Which audio codec to use when the audio stream is being transcoded. Can be one of `mp3`, `aac`, `libopus`.

The default value is `aac`.

### Video Codec (`ffmpeg.targetVideoCodec`) {#ffmpeg.targetVideoCodec}

Which video codec to use when the video stream is being transcoded. Can be one of `h264`, `hevc`, `vp9` or `av1`.

The default value is `h264`.

### Target resolution (`ffmpeg.targetResolution`) {#ffmpeg.targetResolution}

When transcoding a video stream, downscale the largest dimension to this value while preserving aspect ratio. Videos are never upscaled.

The default value is `720`.

### Transcode policy (`ffmpeg.transcode`) {#ffmpeg.transcode}

> The transcoding policy configures which streams of a video asset will be transcoded. The transcoding decision is made independently for video streams and audio streams. This means that if a video stream needs to be transcoded, but an audio stream does not, then the video stream will be transcoded while the audio stream will be copied. If the transcoding policy does not require any stream to be transcoded, then no separate video file will be created.

Reviewer: ```suggestion
The transcoding policy configures which streams of a video asset will be transcoded. The transcoding decision is made independently for video streams and audio streams. This means that if a video stream needs to be transcoded, but an audio stream does not, then the video stream will be transcoded while the audio stream will be copied. If the transcoding policy does not require any stream to be transcoded and does not require the video to be remuxed, then no separate video file will be created.
```
It is actually possible for a video to not be transcoded and simply remuxed.

The default policy is `required`.

#### All videos (`all`) {#ffmpeg.transcode-all}

Videos are always transcoded. This ensures consistency during video playback.

#### Don't transcode any videos (`disabled`) {#ffmpeg.transcode-disabled}

Videos are never transcoded. This saves space and resources on the server, but may prevent playback on devices that don't support the source format (especially web browsers) or result in high bandwidth usage when playing high-bitrate files.

#### Only videos not in an accepted format (`required`) {#ffmpeg.transcode-required}

Video streams are transcoded when any of the following conditions are met:

- The video is HDR.

> - The video is not in the yuv420p colour space.

Reviewer: ```suggestion
- The video is not in the yuv420p pixel format.
```
